File: backend/src/main/java/com/example/shopdb/service/ChatService.java
package com.example.shopdb.service;

import com.example.shopdb.dto.response.ChatResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface ChatService {
    ChatResponse sendMessage(Long senderId, Long receiverId, String message, MultipartFile image) throws Exception;
    List<ChatResponse> getChatBetweenUsers(Long user1Id, Long user2Id);
    void markAsRead(Long chatId);

    // Search chat có phân trang
    Page<ChatResponse> search(Long senderId, Long receiverId, String keyword, Boolean isRead, Pageable pageable);
}

File: frontend-web/src/services/chatService.js
import api from "./api";

// Gửi tin nhắn (có thể kèm file hình)
const sendMessage = async ({ senderId, receiverId, message, imageFile }) => {
  const formData = new FormData();
  formData.append("senderId", senderId);
  formData.append("receiverId", receiverId);
  if (message) formData.append("message", message);
  if (imageFile) formData.append("image", imageFile);

  const response = await api.post("/chats/send", formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });
  return response.data;
};

// Lấy chat giữa 2 user
const getChatBetweenUsers = async (user1Id, user2Id) => {
  const response = await api.get("/chats/between", {
    params: { user1Id, user2Id },
  });
  return response.data;
};

// Đánh dấu chat đã đọc
const markAsRead = async (chatId) => {
  await api.put(`/chats/${chatId}/read`);
};
const searchChats = async ({ senderId, receiverId, keyword, isRead, page = 0, size = 10 }) => {
  const response = await api.get("/chats/search", {
    params: { senderId, receiverId, keyword, isRead, page, size },
  });
  return response.data;
};

// Gán object vào biến trước khi export
const chatService = {
  sendMessage,
  getChatBetweenUsers,
  markAsRead,
  searchChats, 
};

export default chatService;
